extensions/wikia/AdEngine/js/AdEngineRunner.js

```diff
@@ -19,9 +19,11 @@ define('ext.wikia.adEngine.adEngineRunner', [
 	 * @param {function} runAdEngine
 	 */
 	function delayRun(runAdEngine) {
-		var enabledBidders = [],
-			biddersQueue = [],
-			startedByBidders = false;
+		var biddersQueue = [],
+			enabledBidderNames = [],
```

Comment:
I'm not sure I understand @fraszczakszymon's point. The current implementation makes sense to me ;-) I'd just add few more cosmetic changes:
* better describe the `diff()` method,
* use more descriptive parameters names in `diff()`,
* add unit tests to it :japanese_ogre: 

Szymon, can you describe more the magic you mentioned and desire? :-)